### PR TITLE
update python tests to use componentize-py template

### PR DESCRIPTION
**Pull request by karthik2804**
Updated the test based on the latest update based on https://github.com/fermyon/spin-python-sdk/pull/66

Also, updates the testing readme to include information about running all the tests.

### CODE REVIEW

File: tests/README.md [hunk 1]
```diff
@@ -36,3 +36,9 @@ make test-integration
 ```
 
 Note that this also runs the runtime tests as well.
+
+To run all the tests that include external dependencies use:
+
+```bash
+make test-integration-full
+```
```
Comment (rylev) [inline tests/README.md:44]: This is going to conflict with https://github.com/fermyon/spin/pull/2268 which includes a bit more detail

File: tests/integration.rs [hunk 2]
```diff
@@ -417,12 +417,18 @@ Caused by:
     #[cfg(feature = "extern-dependencies-tests")]
     // Ignore until https://github.com/fermyon/spin-python-sdk/pull/65 is merged
```
Comment (rylev) [inline tests/integration.rs:418]: I think this can be removed now

**Review by rylev — APPROVED** [review]

File: tests/README.md (hunk 1 above)
Comment (karthik2804) [reply tests/README.md:44]: I will remove this.

File: tests/integration.rs (hunk 2 above)
Comment (karthik2804) [reply tests/integration.rs:418]: Will do.

File: tests/integration.rs [hunk 3]
```diff
@@ -415,14 +415,25 @@ Caused by:
 
     #[test]
     #[cfg(feature = "extern-dependencies-tests")]
-    // Ignore until https://github.com/fermyon/spin-python-sdk/pull/65 is merged
     fn http_python_template_smoke_test() -> anyhow::Result<()> {
+        let prebuild = |env: &mut testing_framework::TestEnvironment<_>| {
+            let mut setup_venv = std::process::Command::new("python3");
+            setup_venv.args(["-m", "venv", "venv"]);
+            env.run_in(&mut setup_venv);
+            let mut activate_venv = std::process::Command::new("source");
```
Comment (dicej) [inline tests/integration.rs:423]: Does this actually work?  I.e. does it modify the environment of the current process?  I would think it only works inside a shell.

Comment (karthik2804) [reply tests/integration.rs:423]: Good point! I am not sure why the test does not pass in CI though even without it.

**Review by itowlson — APPROVED** [review]

**Comment by karthik2804** [comment]
seem to be running into an unrelated failure
```
failures:
    integration_tests::test_vault_config_provider
test result: FAILED. 36 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 371.07s
```

**Pull request merged**